base/utils/config.go

```diff
@@ -66,59 +66,48 @@ type Config struct {
 }
 
 func init() {
-	initDBFromEnv()
-	initAPIFromEnv()
-	initKafkaFromEnv()
-	initServicesFromEnv()
-	initCloudwatchFromEnv()
-	initPrometheusPushGatewayFromEnv()
+	// topics have to be set first, they may be translated via clowder
+	initTopicsFromEnv()
 	if clowder.IsClowderEnabled() {
 		initDBFromClowder()
 		initAPIromClowder()
 		initKafkaFromClowder()
 		initServicesFromClowder()
 		initCloudwatchFromClowder()
 	}
+	// init non-clowder setting and allow local overwrites
+	initDBFromEnv()
+	initKafkaFromEnv()
+	initServicesFromEnv()
+	initPrometheusPushGatewayFromEnv()
 }
 
 func initDBFromEnv() {
-	Cfg.DBHost = Getenv("DB_HOST", "UNSET")
-	Cfg.DBName = Getenv("DB_NAME", "UNSET")
-	Cfg.DBPort = GetIntEnvOrDefault("DB_PORT", -1)
-	Cfg.DBSslMode = Getenv("DB_SSLMODE", "UNSET")
-	Cfg.DBSslRootCert = Getenv("DB_SSLROOTCERT", "")
-	Cfg.DBAdminUser = Getenv("DB_ADMIN_USER", "")
-	Cfg.DBAdminPassword = Getenv("DB_ADMIN_PASSWD", "")
-	Cfg.DBUser = Getenv("DB_USER", "UNSET")
-	Cfg.DBPassword = Getenv("DB_PASSWD", "UNSET")
+	Cfg.DBHost = Getenv("DB_HOST", Cfg.DBHost)
+	Cfg.DBPort = GetIntEnvOrDefault("DB_PORT", Cfg.DBPort)
+	Cfg.DBSslRootCert = Getenv("DB_SSLROOTCERT", Cfg.DBSslRootCert)
+	Cfg.DBUser = Getenv("DB_USER", Cfg.DBUser)
+	Cfg.DBPassword = Getenv("DB_PASSWD", Cfg.DBPassword)
 	Cfg.DBDebug = GetBoolEnvOrDefault("DB_DEBUG", false)
 	Cfg.DBStatementTimeoutMs = GetIntEnvOrDefault("DB_STATEMENT_TIMEOUT_MS", 0)
 	Cfg.DBMaxConnections = GetIntEnvOrDefault("DB_MAX_CONNECTIONS", 250)
 	Cfg.DBMaxIdleConnections = GetIntEnvOrDefault("DB_MAX_IDLE_CONNECTIONS", 50)
 	Cfg.DBMaxConnectionLifetimeS = GetIntEnvOrDefault("DB_MAX_CONNECTION_LIFETIME_S", 60)
 }
 
-func initAPIFromEnv() {
-	Cfg.PublicPort = GetIntEnvOrDefault("PUBLIC_PORT", -1)
-	Cfg.PrivatePort = GetIntEnvOrDefault("PRIVATE_PORT", -1)
-	Cfg.MetricsPort = GetIntEnvOrDefault("METRICS_PORT", -1)
-	Cfg.MetricsPath = Getenv("METRICS_PATH", "/metrics")
-}
-
 func initKafkaFromEnv() {
-	Cfg.KafkaAddress = Getenv("KAFKA_ADDRESS", "")
-	Cfg.KafkaSslEnabled = GetBoolEnvOrDefault("ENABLE_KAFKA_SSL", false)
-	Cfg.KafkaSslCert = Getenv("KAFKA_SSL_CERT", "")
+	Cfg.KafkaSslCert = Getenv("KAFKA_SSL_CERT", Cfg.KafkaSslCert)
 	Cfg.KafkaSslSkipVerify = GetBoolEnvOrDefault("KAFKA_SSL_SKIP_VERIFY", false)
-	saslType := Getenv("KAFKA_SASL_TYPE", "scram")
-	Cfg.KafkaSaslType = &saslType
-	Cfg.KafkaUsername = Getenv("KAFKA_USERNAME", "")
-	Cfg.KafkaPassword = Getenv("KAFKA_PASSWORD", "")
+	Cfg.KafkaUsername = Getenv("KAFKA_USERNAME", Cfg.KafkaUsername)
+	Cfg.KafkaPassword = Getenv("KAFKA_PASSWORD", Cfg.KafkaPassword)
```

Comment:
do we need to overwrite kafka config for local deployment or tests?

Reply:
We don't do it currently but I assume with new fedramp effort it will be useful.